internal/status/handle.go

```diff
@@ -0,0 +1,72 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/open-telemetry/opentelemetry-operator/internal/manifests"
+	"github.com/open-telemetry/opentelemetry-operator/internal/version"
+	collectorupgrade "github.com/open-telemetry/opentelemetry-operator/pkg/collector/upgrade"
+)
+
+const (
+	eventTypeNormal  = "Normal"
+	eventTypeWarning = "Warning"
+
+	reasonError         = "Error"
+	reasonStatusFailure = "StatusFailure"
+	reasonInfo          = "Info"
+)
+
+// HandleReconcileStatus handles updating the status of the CRDs managed by the operator.
+// TODO: make the status more useful https://github.com/open-telemetry/opentelemetry-operator/issues/1972
+func HandleReconcileStatus(ctx context.Context, log logr.Logger, params manifests.Params, err error) (ctrl.Result, error) {
+	log.V(2).Info("updating collector status")
+	if err != nil {
+		params.Recorder.Event(&params.Instance, eventTypeWarning, reasonError, err.Error())
+		return ctrl.Result{}, err
+	}
+	changed := params.Instance.DeepCopy()
+
+	up := &collectorupgrade.VersionUpgrade{
+		Log:      params.Log,
+		Version:  version.Get(),
+		Client:   params.Client,
+		Recorder: params.Recorder,
+	}
+	upgraded, upgradeErr := up.ManagedInstance(ctx, *changed)
+	if upgradeErr != nil {
+		// don't fail to allow setting the status
+		params.Log.Error(upgradeErr, "failed to upgrade the OpenTelemetry CR")
+	}
+	changed = &upgraded
```

Comment:
@iblancasa this is where i added the logic from your recent PR... In a followup to this we should probably change the upgrade logic to use pointers rather than the way we do it now to keep it in line with the rest of our signatures.